src/server/snapshot.cc

```diff
@@ -44,47 +44,45 @@ void SliceSnapshot::Start(bool include_journal_changes) {
   snapshot_version_ = db_slice_->RegisterOnChange(move(on_change));
   VLOG(1) << "DbSaver::Start - saving entries with version less than " << snapshot_version_;
 
-  if (include_journal_changes) {
+  if (stream_journal) {
     auto* journal = db_slice_->shard_owner()->journal();
     DCHECK(journal);
-    journal_cb_id_ = journal->RegisterOnChange(
-        [this](const journal::Entry& e) { OnJournalEntry(e); });
+    journal_cb_id_ =
+        journal->RegisterOnChange([this](const journal::Entry& e) { OnJournalEntry(e); });
   }
 
   sfile_.reset(new io::StringFile);
-
   rdb_serializer_.reset(new RdbSerializer(sfile_.get()));
 
-  fb_ = fiber([this] {
-    FiberFunc();
+  snapshot_fb_ = fiber([this, stream_journal] {
+    SerializeEntriesFb();
+    if (!stream_journal) {
+      CloseRecordChannel();
```

Comment:
you call it twice: here and in `Stop()`. Why?

Reply:
The code flow is a bit unobvious. Stop only needs to be called if the journal is enabled, otherwise the snapshot closes automatically right here

Reply:
I've also cut the number of possible use options - currently enabling journaling means not stopping after serialization

Reply:
Maybe worth adding comments explaining this next time you change this file